ipyvizzu.py

```diff
@@ -139,24 +129,53 @@ class Chart:
 
     VIZZU = "https://cdn.jsdelivr.net/npm/vizzu@latest/dist/vizzu.min.js"
 
+    _INITIALIZE = """
+        <div id="myVizzu_{id}" style="width:{div_width}; height:{div_height};"/>
+        <script type="module">
+        import Vizzu from '{vizzu}';
+        document.chart_{id} = new Vizzu('myVizzu_{id}').initializing;
+        </script>
+    """
+
+    _ANIMATE = """
+        <script type="module">
+        document.chart_{id}.then( chart => {{{animation}}});
```

Comment:
Couldn't we use a local variable instead of the global 'document'?

Reply:
global document used here, because Vizzu can only be imported statically to module type scripts, and module type scripts are using separate namespaces. They do not see each other. 

An alternative would be to use async dynamic import. Then the code would be something like this:
```html
<script>
let chart_{id} = import('{vizzu}').then(Vizzu => new Vizzu.default('myVizzu_{id}').initializing;
</script>

<script>
chart_{id}.then( chart => {{{animation}}});
</script>
```
Actually, I think this would be better.

Reply:
I think the generated javascript code should look like if someone wrote it by hand. That's why the multiple `<script />` block seemed strange to me.

In the vizzu.js [documentation](https://lib.vizzuhq.com/0.4/#chapter-0.17) the user just call `store` and `animate`:
```javascript
snapshot = chart.store();

chart.animate(...);

chart.animate(snapshot);
```

Why do we need this extra round with auto generated snapshot and restore?
Is it not possible to define a variable from the result of `store` in a `chart.then( chart => {...} )`


It would be good if the python API could follow the same semantic:

```python
snapshot = chart.store()
chart.animate(snapshot)
```

Something like https://github.com/vizzuhq/ipyvizzu/pull/47

Reply:
Using more script because we d like to animate the same chart which is created in the first notebook cell in every cell in the notebook.

#47 is a good suggestion

Reply:
Sorry I missunderstod the feature.

If I get it right you want to save the state of the chart so the chart of the next notebook cell can be initiated from the previous one. In this way the test data can be loaded only once, because the next cells can use the snapshot directly.

Reply:
Could we split only the vizzu import  (and the global snapshot store) into a separate script node?

Ipyvizzu would generate the import script node only once, as an initialize phase, and all the regular script node would reach vizzu from there. The plotly [HtmlRender](https://github.com/plotly/plotly.py/blob/1788c1c95e9df9710454ea7a0285dc5a0b165eee/packages/python/plotly/plotly/io/_base_renderers.py#L271) has a similar strategy.

Reply:
> Sorry I missunderstod the feature.
> 
> If I get it right you want to save the state of the chart so the chart of the next notebook cell can be initiated from the previous one. In this way the test data can be loaded only once, because the next cells can use the snapshot directly.

Yes, the main goal is to enable the ipyvizzu api to support upper level functionalities like anim control (it can be in an extension, but the api should support the base). For these we need to "interact" the chart and not enough to display it after every code complete.

Reply:
> Could we split only the vizzu import (and the global snapshot store) into a separate script node?
> 
> Ipyvizzu would generate the import script node only once, as an initialize phase, and all the regular script node would reach vizzu from there. The plotly [HtmlRender](https://github.com/plotly/plotly.py/blob/1788c1c95e9df9710454ea7a0285dc5a0b165eee/packages/python/plotly/plotly/io/_base_renderers.py#L271) has a similar strategy.

Of course, this draft is still evolving, I submitted it so early because of suggestions. By the end, when it will be closed, I hope it will be as simple and good as possible.

Reply:
Does the snapshot feature interfere with animation control?

At the first sight it seemed to be a different issue (a very complex one). I thought the main problem with snapshot feature is that how to share snapshot between notebook cells. Am i missing something?

Reply:
@nyirog We discuss it with @simzer and in the current stage of #46 we reached the desired functionality in ipyvizzu.
(Obviously I have to follow up unit tests and docs)

Reply:
Sounds good.

The multiple script block seemed strange at first, but it is a good solution if store/snapshot works between the notebook cell .